sbnana/CAFAna/Systs/EnergySysts.h

```diff
@@ -0,0 +1,240 @@
+//ETW May 2020 Fresh version for SBN
+
+#pragma once
+
+#include "sbnana/CAFAna/Core/ISyst.h"
+#include "sbnana/CAFAna/Core/Utilities.h"
+#include "sbnana/CAFAna/Analysis/ExpInfo.h"
+#include "sbnanaobj/StandardRecord/Proxy/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+
+namespace ana
+{
+
+  enum class EnergyScaleSystTerm {
+    Constant,
+    Sqrt,
+    InverseSqrt
+  };
+
+  enum class EnergyScaleSystParticle {
+    Muon,
+    Hadron,
+    Neutron,
+    EM,
+    ChargedHadron
+  };
+
+  enum class EnergyScaleSystDetector {
+    SBND,
+    MicroBooNE,
+    ICARUS,
+    All
+  };
+
+  template<EnergyScaleSystTerm term, EnergyScaleSystParticle part, EnergyScaleSystDetector detector = EnergyScaleSystDetector::All>
+  class EnergyScaleSyst: public ISyst
+  {
+  public:
+    EnergyScaleSyst(double _uncertainty, std::string name) : 
```

Comment:
```suggestion
    EnergyScaleSyst(double _uncertainty, const std::string& name) : 
```

Reply:
I think we also want to allow a separate shortName and latexName. The names you have now look mostly like shortNames to me.

Reply:
Added latex names to all the systs in EnergySysts.cxx. The names are long and pretty verbose. If you have suggestions for better names I can change those.